### PR TITLE
Fix hspec

### PR DESCRIPTION
**Pull request by juliens**
This PR fixes the 4.* and 5.* specification.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -220,7 +225,7 @@ func (sc *serverConn) handleStreams() {
 				continue
 			}
 
-			if len(strms) >= int(sc.st.maxStreams) {
+			if len(strms)+len(closedStrms) >= int(sc.st.maxStreams) {
```
Comment (dgrr) [inline server.go:228]: len(closedStrms)? Why do we keep the closedStrms? That can grow infinitely

Comment (juliens) [reply server.go:228]: We need to be able to identify closed stream to send a closed stream error when frame arrive on those streams.

Comment (dgrr) [reply server.go:228]: Ok, I am fine with that, but the maxStreams is the maximum current open streams, right? So, if we do `current_streams + closed_streams >= max_streams` it's going to fail as soon as the client opened more than 128 streams, although the client might have only one stream open at that time.

Comment (juliens) [reply server.go]: Yes, You're right, the fix was not the right one but a consequence of another bug around `SETTINGS_INITIAL_WINDOW_SIZE`.

So I remove it and I will do another PR with the fix on `SETTINGS_INITIAL_WINDOW_SIZE` to fix this spec.

**Comment by dgrr** [comment]
Is it ready to merge?

**Comment by juliens** [comment]
Yes, still have a fix to do for the 5.1.2 spec but, I will do it in another PR.

**Comment by dgrr** [comment]
Thanks for your PR!

**Pull request merged**